### PR TITLE
Adding Brandfolder app

### PR DESCRIPTION
**Pull request by schwindtkevin**

### CODE REVIEW

File: apps/brandfolder/src/index.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        const att_id = payload.attachments[0].id
+        const attachment = payload.included.find(att => att.id === att_id)
+        if (attachment){
+          sdk.close([attachment])
+        }
+      }
+    }
+  });
+
+  // TODO maybe figure out how to make this work
```
Comment (lehnerm) [inline apps/brandfolder/src/index.js]: What were you trying to solve here?

File: apps/brandfolder/src/index.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    allowHeightOverflow: true,
+  });
+
+  if (!Array.isArray(result)) {
+    return [];
+  }
+
+  return result
+
+  // return result.map(item => ({
```
Comment (lehnerm) [inline apps/brandfolder/src/index.js]: Does the resulting data contain any personally identifiable information?

File: apps/brandfolder/src/index.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import { setup } from 'shared-dam-app';
+
+const CTF_APP_URL = 'https://integration-contentful-app.brandfolder-svc.com'
```
Comment (lehnerm) [inline apps/brandfolder/src/index.js]: For the public marketplace app, the hosting will be done by us at https://brandfolder.ctfapps.net

File: apps/brandfolder/src/index.js (hunk 1 above)
Comment (schwindtkevin) [reply apps/brandfolder/src/index.js]: Other apps here add the event listener to `document` but I only managed to get it to work on `window`

File: apps/brandfolder/src/index.js (hunk 3 above)
Comment (schwindtkevin) [reply apps/brandfolder/src/index.js]: Updating, thanks!

File: apps/brandfolder/src/index.js (hunk 2 above)
Comment (schwindtkevin) [reply apps/brandfolder/src/index.js]: No. An example result is ```[
    {
        "id": "pfbfh7-f4zem0-dpcdo2",
        "type": "attachments",
        "mimetype": "image/png",
        "filename": "brandfolder-icon.png",
        "size": 9252,
        "width": 312,
        "height": 294,
        "url": "https://s3.amazonaws.com/bf.boulder.prod/pfbfh7-f4zem0-dpcdo2/original/brandfolder-icon.png",
        "thumbnail_url": "https://assets.brandfolder.com/pfbfh7-f4zem0-dpcdo2/element.png?v=1555005228",
        "position": 0,
        "included": null,
        "asset": {
            "id": "6skcfbkq"
        }
    }
]```

File: apps/brandfolder/src/index.js (hunk 1 above)
Comment (davidfateh) [reply apps/brandfolder/src/index.js]: Make sense, we can probably clean up the commented out code then.

File: apps/brandfolder/src/index.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  window.addEventListener('message', e => {
+    const { data, origin } = e
+    if (origin === 'https://integration-panel-ui.brandfolder-svc.com'){
+      const { event, payload } = data;
+      console.log('Brandfolder event', event)
+      if (event == 'selectedAttachment') {
+        sdk.close([payload]);
+      }
+      else if (data.event === 'selectedAsset') {
+        const att_id = payload.attachments[0].id
```
[the remark this answers is not in the document]
Comment (schwindtkevin) [reply apps/brandfolder/src/index.js]: I don't think so, but I've added a check to be safe.

**Comment by lehnerm** [comment]
@schwindtkevin Can you please rebase?

**Comment by schwindtkevin** [comment]
> @schwindtkevin Can you please rebase?

@lehnerm Sorry, just noticed this. I had clicked the button to merge the latest changes from master. Is that sufficient?

File: apps/brandfolder/src/index.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  //     "height": 294,
+  //     "url": "https://s3.amazonaws.com/bf.boulder.prod/pfbfh7-f4zem0-dpcdo2/original/brandfolder-icon.png",
+  //     "thumbnail_url": "https://assets.brandfolder.com/pfbfh7-f4zem0-dpcdo2/element.png?v=1555005228",
+  //     "position": 0,
+  //     "included": null,
+  //     "asset": {
+  //         "id": "6skcfbkq"
+  //     }
+  // }]
+  return result
```
Comment (giotiskl) [inline apps/brandfolder/src/index.js]: For safety purposes (not to accidentally leak PII in the future) in all the other we are whitelisting the fields of the response that we want persisted on the Contentful side (e.g. https://github.com/contentful/apps/blob/master/apps/cloudinary/src/index.js#L12 if you check how the `FIELDS_TO_PERSIST` constant is used).

Could we implement the same check here?

**Pull request merged**